src/managers/GuildScheduledEventManager.js

```diff
@@ -0,0 +1,247 @@
+const { PrivacyLevels, GuildScheduledEventEntityTypes, GuildScheduledEventStatuses } = require('../util/Constants');
+
+/**
+ * Manages API methods for GuildScheduledEvents and stores their cache.
+ * @extends {CachedManager}
+ */
+class GuildScheduledEventManager extends CachedManager {
+  constructor(guild, iterable) {
+    super(guild.client, GuildScheduledEvent, iterable);
+
+    /**
+     * The guild this manager belongs to
+     * @type {Guild}
+     */
+    this.guild = guild;
+  }
+
+  /**
+   * The cache of this manager
+   * @type {Collection<Snowflake, GuildScheduledEvent>}
+   * @name GuildScheduledEventManager#cache
+   */
+
+  /**
+   * Data that resolves to give a GuildScheduledEvent object. This can be:
+   * * A Snowflake
+   * * A GuildScheduledEvent object
+   * @typedef {Snowflake|GuildScheduledEvent} GuildScheduledEventResolvable
+   */
+
+  /**
+   * Options used to create a guild scheduled event.
+   * @typedef {Object} GuildScheduledEventCreateOptions
+   * @property {string} name The name of the guild scheduled event
+   * @property {DateResolvable} scheduledStartTime The time to schedule the event at
+   * @property {DateResolvable} [scheduledEndTime] The time to end the event at
+   * <warn>This is required if entityType is 'EXTERNAL'</warn>
+   * @property {PrivacyLevel|number} privacyLevel The privacy level of the guild scheduled event
+   * @property {GuildScheduledEventEntityType|number} entityType The scheduled entity type of the event
+   * @property {string} [description] The description of the guild scheduled event
+   * @property {GuildVoiceChannelResolvable} [channel] The channel of the guild scheduled event
+   * @property {string} [location] The location of the guild scheduled event
+   */
+
+  /**
+   * Creates a new guild scheduled event.
+   * @param {GuildScheduledEventCreateOptions} options Options for creating the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   */
+  async create(options) {
+    if (typeof options !== 'object') throw new TypeError('INVALID_TYPE', 'options', 'object', true);
+    let { privacyLevel, entityType, channel, location, name, scheduledStartTime, description, scheduledEndTime } =
+      options;
+
+    if (typeof privacyLevel === 'string') privacyLevel = PrivacyLevels[privacyLevel];
+    if (typeof entityType === 'string') entityType = GuildScheduledEventEntityTypes[entityType];
+    const channelId = this.guild.channels.resolveId(channel);
+
+    const data = await this.client.api.guilds(this.guild.id, 'scheduled-events').post({
+      data: {
+        channel_id: channelId,
+        name,
+        privacy_level: privacyLevel,
+        scheduled_start_time: new Date(scheduledStartTime).toISOString(),
+        scheduled_end_time: scheduledEndTime ? new Date(scheduledEndTime).toISOString() : undefined,
+        description,
+        entity_type: entityType,
+        entity_metadata: {
+          location,
+        },
```

Comment:
Do we want to abstract it like this? CC @discordjs/the-big-4

Reply:
Probably not? I think we should stick to being close with Discord.

We could also make a builder for this, and since builders need to be very close to Discord's types, it can be very odd if we mix the two, as well as making both interfaces incompatible.

Reply:
@iCrawl thoughts?

Reply:
I don't think we need to.